src/main/java/in/projecteka/consentmanager/user/UpdateUserRequestValidator.java

```diff
@@ -0,0 +1,56 @@
+package in.projecteka.consentmanager.user;
+
+import com.google.common.base.Strings;
+import io.vavr.control.Validation;
+import org.passay.CharacterRule;
+import org.passay.CharacterSequence;
+import org.passay.EnglishCharacterData;
+import org.passay.IllegalSequenceRule;
+import org.passay.LengthRule;
+import org.passay.PasswordData;
+import org.passay.PasswordValidator;
+import org.passay.RuleResult;
+import org.passay.RuleResultDetail;
+import org.passay.SequenceData;
+
+import java.util.Arrays;
+
+import static java.lang.String.format;
+
+public class UpdateUserRequestValidator {
+    public static Validation<String, String> validatePassword(String password) {
+        if (Strings.isNullOrEmpty(password)) {
+            return Validation.invalid("password can't be empty");
+        }
+        PasswordValidator validator = new PasswordValidator(Arrays.asList(
+                new LengthRule(8, 30),
+                new CharacterRule(EnglishCharacterData.UpperCase, 1),
+                new CharacterRule(EnglishCharacterData.LowerCase, 1),
+                new CharacterRule(EnglishCharacterData.Digit, 1),
+                new CharacterRule(EnglishCharacterData.Special, 1),
+                new IllegalSequenceRule(new SequenceData() {
+                    @Override
+                    public String getErrorCode() {
+                        return "cannot have three or more consecutive numbers";
+                    }
+
+                    @Override
+                    public CharacterSequence[] getSequences() {
+                        return new CharacterSequence[]{
+                                new CharacterSequence("`1234567890-=")
+                        };
+                    }
+                }, 3, false)));
+        RuleResult result = validator.validate(new PasswordData(password));
+        if (result.isValid()) {
+            return Validation.valid(password);
+        }
+        var error = result.getDetails()
+                .stream()
+                .map(RuleResultDetail::toString)
+                .reduce((left, right) -> format("%s, %s", left, right))
+                .map(message -> format("password has following issues: %s", message))
+                .orElse("password did not meet criteria");
+        return Validation.invalid(error);
+    }
```

Comment:
There is already a password validator. We can use that.

Reply:
It's in `SignupRequestValidator.java`

Reply:
Validators are specific to the request

Reply:
Yes. I get that. But since it is same as it is for the other request we can reuse that. If it changes in future then we probably can change it.

Reply:
Done.